src/components/SurvivorSelection.tsx

```diff
@@ -0,0 +1,40 @@
+import "../styles/survivor.css";
+import { Survivor, UserSelection } from "../utilities/types";
+
+export interface Props {
+  selectedSurvivor: Survivor;
+  survivorList: Survivor[];
+  setSelectedSurvivor: React.Dispatch<React.SetStateAction<Survivor>>;
+  setUserSelection: React.Dispatch<React.SetStateAction<UserSelection>>;
+}
+
+const SurvivorSelection = ({
+  selectedSurvivor,
+  survivorList,
+  setSelectedSurvivor,
+  setUserSelection
+}: Props) => {
```

Comment:
purely styling / preference, but i found this to be easier + cleaner -- UNLESS you need to export the interface (i see you put export but didnt actually use it anywhere else)

the below is basically as setting up the prop types, but within the FC in-house instead of setting it up then pulling it in

```suggestion
const SurvivorSelection: React.FC<{
  selectedSurvivor: Survivor;
  survivorList: Survivor[];
  setSelectedSurvivor: React.Dispatch<React.SetStateAction<Survivor>>;
  setUserSelection: React.Dispatch<React.SetStateAction<UserSelection>>;
}> = (
  selectedSurvivor,
  survivorList,
  setSelectedSurvivor,
  setUserSelection
) => {
```

Reply:
![image](https://github.com/JeffreyChuCPA/Risk-of-Rain-2-Multishop-Simulator/assets/141854177/20d95990-63ed-45b2-8dbc-41f9d91b37b1)
Making changes brings up error for the props being still considered an any type

Reply:
oops, typo on my part

```suggestion
const SurvivorSelection: React.FC<{
  selectedSurvivor: Survivor;
  survivorList: Survivor[];
  setSelectedSurvivor: React.Dispatch<React.SetStateAction<Survivor>>;
  setUserSelection: React.Dispatch<React.SetStateAction<UserSelection>>;
}> = ({
  selectedSurvivor,
  survivorList,
  setSelectedSurvivor,
  setUserSelection
}) => {
```

Reply:
So i can either explicitly define an interface (without adding export) OR pass as props as a generic to the component function? what is really the benefit of one over the other